subprojects/core/source/jcli/core/flags.d

```diff
@@ -313,22 +267,88 @@ private template _ArgFlagsInfo(Flags)
         return Flags.max.bitsSet.array[$ - 1] + 1;
     }();
 
-    Flags[argFlagCount] getFlagsInfo(TUDA)()
+    Flags[argFlagCount] getFlagsInfo(alias f)()
     {
         Flags[argFlagCount] result;
 
         static foreach (memberName; __traits(allMembers, Flags))
         {{
             size_t index = __traits(getMember, Flags, memberName).bitsSet.front;
-            static foreach (uda; __traits(getAttributes, __traits(getMember, Flags, memberName)))
-            {
-                static if (is(typeof(uda) == TUDA))
-                {
-                    result[index] = uda.value;
-                }
-            }
+            ArgFlags flag = f(cast(ArgFlags) (1 << index));
+            result[index] = flag;
         }}
 
         return result;
     }
 }
+
+
+ArgFlags getIncompatible(ArgFlags flags) pure nothrow @nogc @safe
+{
+    switch (flags) with (ArgFlags)
+    {
+        case _parseAsFlagBit:
+            return _multipleBit;
+        case _canRedefineBit:
+            return _countBit;
+        case _repeatableNameBit:
+            return _parseAsFlagBit | _canRedefineBit;
+        case _aggregateBit:
+            return _parseAsFlagBit | _countBit | _canRedefineBit;
+        case _requiredBit:
+            return _optionalBit | _parseAsFlagBit;
+        case _positionalArgumentBit:
+            return _multipleBit | _parseAsFlagBit | _countBit | _canRedefineBit | _repeatableNameBit | _aggregateBit;
+        case _namedArgumentBit:
+            return _positionalArgumentBit;
+        default:
+            break;
+    }
+
+    return ArgFlags.none;
+}
+
+ArgFlags getRequiresAll(ArgFlags flags) pure nothrow @nogc @safe
+{
+    switch (flags) with (ArgFlags)
+    {
+        case _parseAsFlagBit:
+            return _optionalBit;
+        case _canRedefineBit:
+            return _multipleBit;
+        case _repeatableNameBit:
+            return _countBit;
+        case _mayChangeOptionalityWithoutBreakingThingsBit:
+            return _inferedOptionalityBit;
+        default:
+            break;
+    }
+
+    return ArgFlags.none;
+}
+
+ArgFlags getRequiresExactlyOne(ArgFlags flags) pure nothrow @nogc @safe
+{
+    switch (flags) with (ArgFlags)
+    {
+        case _countBit:
+            return _multipleBit | _repeatableNameBit;
+        default:
+            break;
+    }
+
+    return ArgFlags.none;
+}
+
+ArgFlags getRequiresOneOrMore(ArgFlags flags) pure nothrow @nogc @safe
+{
+    switch (flags) with (ArgFlags)
+    {
+        case _multipleBit:
+            return _countBit | _canRedefineBit | _aggregateBit;
+        default:
+            break;
+    }
+
+    return ArgFlags.none;
+}
```

Comment:
I think this is backwards here:

In the previous code `_multipleBit` is marked with `RequiresExactlyOneOf` and `_countBit` is marked with `RequiresOneOrMoreOf`.

Reply:
Apologies for the oversight. I didn't notice that the implementations were swapped. I believe everything else should be fine now.